substrate/frame/asset-conversion/ops/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+			Self::AccountId,
+			AssetId = Self::AssetKind,
+			Balance = <Self::DepositAsset as FungibleInspect<Self::AccountId>>::Balance,
+		>;
+
+		/// Retrieves information about an existing deposit for a given account ID and asset from
+		/// the [`pallet_asset_conversion::Config::PoolAssets`] registry and can initiate the
+		/// refund.
+		type PoolAssetsRefund: Refund<
+			Self::AccountId,
+			AssetId = Self::PoolAssetId,
+			Balance = <Self::DepositAsset as FungibleInspect<Self::AccountId>>::Balance,
+		>;
+
+		/// Means to reset the team for assets from the
+		/// [`pallet_asset_conversion::Config::PoolAssets`] registry.
+		type PoolAssetsTeam: ResetTeam<Self::AccountId, AssetId = Self::PoolAssetId>;
+
+		/// Registry of an asset used as an account deposit for the
+		/// [`pallet_asset_conversion::Config::Assets`] and
+		/// [`pallet_asset_conversion::Config::PoolAssets`] registries.
+		type DepositAsset: FungibleMutate<Self::AccountId>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	// Pallet's events.
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Indicates that a pool has been migrated to the new account ID.
+		MigratedToNewAccount {
+			/// Pool's ID.
+			pool_id: T::PoolId,
+			/// Pool's prior account ID.
+			prior_account: T::AccountId,
+			/// Pool's new account ID.
+			new_account: T::AccountId,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Provided asset pair is not supported for pool.
+		InvalidAssetPair,
+		/// The pool doesn't exist.
+		PoolNotFound,
+		/// Pool's balance cannot be zero.
+		ZeroBalance,
+		/// Indicates a partial transfer of balance to the new account during a migration.
+		PartialTransfer,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			// TODO
+		}
+	}
+
+	/// Pallet's callable functions.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Migrates an existing pool to a new account ID derivation method for a given asset pair.
+		/// If the migration is successful, transaction fees are refunded to the caller.
+		///
+		/// Must be signed.
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::migrate_to_new_account())]
+		pub fn migrate_to_new_account(
+			origin: OriginFor<T>,
+			asset1: Box<T::AssetKind>,
+			asset2: Box<T::AssetKind>,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+
+			let pool_id = T::PoolLocator::pool_id(&asset1, &asset2)
+				.map_err(|_| Error::<T>::InvalidAssetPair)?;
+			let info = Pools::<T>::get(&pool_id).ok_or(Error::<T>::PoolNotFound)?;
+
+			let (prior_account, new_account) =
+				Self::addresses(&pool_id).ok_or(Error::<T>::InvalidAssetPair)?;
+
+			let (asset1, asset2) = pool_id.clone();
+
+			// Assets that must be transferred to the new account id.
+			let balance1 = T::Assets::balance(asset1.clone(), &prior_account);
```

Comment:
We should use `reducible_balance` here instead, same for usage below.

Reply:
If all `balance` is not reducible, I want it to fail, because it is not expected. So basically `balance` must be equal to `reducible_balance`